.github/workflows/ci.yml

```diff
@@ -141,12 +141,22 @@ jobs:
           echo "GOMEMLIMIT=6GiB" >> "$GITHUB_ENV"
           echo "GOGC=80" >> "$GITHUB_ENV"
 
-      - name: run tests without race detection
+      - name: run tests without race detection and path state scheme
         if: matrix.test-mode == 'defaults'
+        env:
+          TEST_STATE_SCHEME: path
         run: |
           packages=`go list ./...`
           stdbuf -oL gotestsum --format short-verbose --packages="$packages" --rerun-fails=1 --no-color=false -- ./... -coverprofile=coverage.txt -covermode=atomic -coverpkg=./...,./go-ethereum/... -timeout 20m -parallel=8 > >(stdbuf -oL tee full.log | grep -vE "INFO|seal")
 
+      - name: run tests without race detection and hash state scheme
```

Comment:
Just noting it here, we can add it later when we make some new modification to hashdb:

I wonder if we shouldn't also run the race tests for both schemes after all. That would give us extra coverage for detecting race issues in our patches to `hashdb` (e.g. changes to the garbage collection of dirty tire nodes for snap sync support) while we still need to race test `pathdb` (e.g. if adding "dynamic" StateHistory threshold turns out to be more complex, or if we would need to do some modification to support snap sync).

Currently, because some of the tests are always running with Hash Scheme, at least few tests are running with race detection and Hash Scheme, so we should be good for now.

Reply:
For peace of mind I added a step to run race tests with hash scheme, and another with path scheme.